activerecord/lib/arel/nodes/node.rb

```diff
@@ -2,8 +2,101 @@
+    # = Using +Arel::Nodes::Node+
+    #
+    # Active Record uses Arel to compose SQL statements. Instead of building SQL strings directly, it's building an
+    # abstract syntax tree (AST) of the statement using various types of Arel::Nodes::Node. Each node represents a
+    # fragment of a SQL statement.
+    #
+    # The intermediate representation allows Arel to compile the statement into the database's specific SQL dialect
+    # only before sending it without having to care about the nuances of each database when building the statement.
+    # It also allows easier composition of statements without having to resort to (brittle) string manipulation.
+    #
+    # == Building constraints
+    #
+    # One of the most common use cases of Arel is generating constraints for +SELECT+ statements. To help with that,
+    # most nodes include a couple of useful factory methods to create subtree structures for common constraints. For
+    # a full list of those, please refer to Arel::Predications.
+    #
+    # The following example creates a equality constraint where the value of the name column on the users table
+    # matches the value DHH.
+    #
+    #   users = Arel::Table.new(:users)
+    #   constraint = users[:name].eq("DHH")
+    #
+    #   # => Arel::Nodes::Equality.new(
+    #   #      Arel::Attributes::Attribute.new(users, "name"),
+    #   #      Arel::Nodes::Casted.new(
+    #   #        "DHH",
+    #   #        Arel::Attributes::Attribute.new(users, "name")
+    #   #      )
+    #   #    )
+    #
+    #   constraint.to_sql # => "users"."name" = 'DHH'
+    #
+    # The constraint fragments can be used with regular ActiveRecord::Relation objects instead of as Hash. The
+    # following two example shows two ways of creating the same query.
+    #
+    #   User.where(name: 'DHH')
+    #
+    #   # => SELECT "users".* FROM "users" WHERE "users"."name" = 'DHH'
+    #
+    #   users = User.arel_table
+    #
+    #   User.where(users[:name].eq('DHH'))
+    #
+    #   # => SELECT "users".* FROM "users" WHERE "users"."name" = 'DHH'
+    #
+    # == Functions
+    #
+    # Arel comes with built in support for SQL functions like +COUNT+, +SUM+, +MIN+, +MAX+, and +AVG+. The
+    # Arel::Expressions module includes factory methods for the default functions.
+    #
+    #   employees = Employee.arel_table
+    #
+    #   Employee.select(employees[:department_id], employees[:salary].average).group(employees[:department_id])
+    #
+    #   # => SELECT "employees"."department_id", AVG("employees"."salary")
+    #   #      FROM "employees" GROUP BY "employees"."department_id"
+    #
+    # It’s also possible to use custom functions by using the Arel::Nodes::NamedFunction node type. It accepts a
+    # function name and an array of parameters.
+    #
+    #   Arel::Nodes::NamedFunction.new('date_trunc', [Arel::Nodes.build_quoted('day'), User.arel_table[:created_at]])
+    #
+    #   # => date_trunc('day', "users"."created_at")
```

Comment:
```suggestion
date_trunc.to_sql => "date_trunc('day', "users"."created_at")"
```

or similar to represent output

Reply:
I think calling `to_sql` on nodes themselves isn't really recommended, that's why I didn't explicitly add the method call. 

Maybe we can make it clearer there this isn't the return value but the representation of the SQL equivalent in Arel by not using `# =>`?

Reply:
Yeah, agree on not wanting to call `to_sql` here: it doesn't know where it's going to be run, so has to fall back to the global connection. More generally, I'd aim to avoid situations where [apart from this sort of casual inspection] anyone feels the need to call `to_sql`... ideally Arel nodes should be drop-in usable in all relevant contexts as-is. (Though perhaps needing to switch some surrounding `#{}` to an `Arel.sql` + placeholder.)

@zzak I think there's been some recent discussion on standardising how we represent "will produce/run this query" vs literally "will return this value" -- am I remembering right that you were in that conversation? (Possibly we just settled on a loose "`=>` means 'gives', and specifics are down to context"?)

Reply:
I removed the `=>` from all SQL examples, following the examples in [`ActiveRecord::Associations`](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/associations.rb) and [`ActiveRecord::Relation`](https://github.com/rails/rails/blob/main/activerecord/lib/active_record/relation.rb)

Reply:
Following up here, I think the conversation comes from #47277 where we discussed removing `=>` from SQL examples. I will try to get this codified in the docs. :pray:

Reply:
cc #47757